File: kitchen-server/src/main/java/com/mykyda/kitchenserver/database/repository/OrderRepository.java
package com.mykyda.kitchenserver.database.repository;

import com.mykyda.kitchenserver.database.entity.Order;
import com.mykyda.kitchenserver.database.enums.OrderStatus;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface OrderRepository extends JpaRepository<Order, UUID> {

    List<Order> getOrdersByStatusNotAndFacilityIdOrderByCreatedAtAsc(OrderStatus orderStatus, UUID facilityId);
}

File: pizza-server/src/main/kotlin/com/acheron/pizzaserver/repo/OrderRepository.kt
package com.acheron.pizzaserver.repo

import com.acheron.pizzaserver.entity.Order
import com.acheron.pizzaserver.entity.OrderStatus
import com.acheron.pizzaserver.entity.Pizza
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.UUID
@Repository
interface OrderRepository : CrudRepository<Order, UUID> {
    fun findAll(pageable: Pageable): Page<Order>

    fun findAllByStatus(status: OrderStatus, pageable: Pageable): Page<Order>

    fun findAllByUserIdAndStatus(
        userId: Long,
        status: OrderStatus,
        pageable: Pageable
    ): Page<Order>

    fun findAllByUserId(userId: Long, pageable: Pageable): Page<Order>



}